src/Framing/AmqpValue.cs

```diff
@@ -89,7 +89,7 @@ internal override void EncodeValue(ByteBuffer buffer)
             }
             else
             {
-                this.WriteValue(buffer, this.value);
+                this.WriteValue(buffer, this.Value);
```

Comment:
I think you should fix all the usages of `this.value`.  Without it, the condition above also will not work correctly.